### PR TITLE
Use find_package for OpenSSL detection

### PR DESCRIPTION
**Pull request by JosiahWI**
This refactors CMakeLists.txt to call find_package() instead of using our own detection logic. The result is easier to read, and will benefit from improvements to the FindOpenSSL module, which is battle-hardened.

The following options' behavior was preserved:
DEPS_PATH, OPENSSL_INCLUDE_PATH, and OPENSSL_LIBRARY_PATH.

The change was implemented on OS X, so the developer was able to verify that Homebrew's OpenSSL1.1 can be detected after the change.

This is continuing the work started in #358.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -103,68 +103,55 @@ else ()
...
+    endif()
+
+    # FindOpenSSL does not detect OpenSSL installed by homebrew.
+    if (APPLE)
+        list(APPEND CMAKE_LIBRARY_PATH ${OPENSSL_LIBRARY_PATH} /opt/homebrew/opt/openssl@1.1)
+        list(APPEND CMAKE_INCLUDE_PATH ${OPENSSL_INCLUDE_PATH} /opt/homebrew/opt/openssl@1.1)
+    endif ()
+
+    # This needs to be mantained for backwards compatibility.
+    set(OPENSSL_ROOT_DIR ${DEPS_PREFIX})
```
Comment (greensky00) [inline CMakeLists.txt:130]: Please put
```
set(OPENSSL_USE_STATIC_LIBS TRUE)
```

**Review by greensky00 — COMMENTED** [review]
Thanks, a few minor comments.

File: CMakeLists.txt [hunk 2]
```diff
@@ -103,68 +103,55 @@ else ()
...
+
+    find_package(OpenSSL QUIET)
+
+    if (OPENSSL_LIBRARY_PATH)
```
[the remark this answers is not in the document]
Comment (JosiahWI) [reply CMakeLists.txt:134]: The purpose of this logic wasn't very clear, but it looked like it was supposed to print out the paths only if the path was given on the command line via `OPENSSL_LIBRARY_PATH`. In the original code, the logic was inverted: it checked `if (NOT OPENSSL_LIBRARY_PATH)`, but I put the positive condition first to make the logic easier to read.

File: CMakeLists.txt [hunk 3]
```diff
@@ -103,68 +103,55 @@ else ()
...
-            set(LIBCRYPTO ${OPENSSL_LIBRARY_PATH}/libcrypto.a)
         endif ()
+    endif ()
 
-        if (OPENSSL_INCLUDE_PATH)
-            message(STATUS "Open SSL include path: ${OPENSSL_INCLUDE_PATH}")
-            include_directories(BEFORE ${OPENSSL_INCLUDE_PATH})
-        endif ()
+    if (OPENSSL_INCLUDE_PATH)
+        message(STATUS "Open SSL include path: ${OPENSSL_INCLUDE_PATH}")
```
[the remark this answers is not in the document]
Comment (JosiahWI) [reply CMakeLists.txt:151]: Thanks, good catch!

File: CMakeLists.txt (hunk 2 above)
[the remark this answers is not in the document]
Comment (JosiahWI) [reply CMakeLists.txt:135]: I was careful to build both ways and compare the old and new behavior to make sure it did not change, but that does not exclude the possibility that the old behavior had a bug. Maybe we should print out the paths unconditionally, or only when the library path was not given explicitly.

[the remark this answers is not in the document]
Comment (greensky00) [reply CMakeLists.txt:135]: Yes, please make it print out the paths regardless of `OPENSSL_LIBRARY_PATH`. Maybe:
```
    if (OpenSSL_FOUND)
        set(LIBSSL OpenSSL::SSL)
        set(LIBCRYPTO OpenSSL::Crypto)
        message(STATUS "OpenSSL library path: ${OPENSSL_SSL_LIBRARY}")
    else ()
        message(STATUS "Use system's side OpenSSL library")
        set(LIBSSL ssl)
        set(LIBCRYPTO crypto)
    endif ()
```

**Review by greensky00 — APPROVED** [review]
LGTM

**Pull request merged**